lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,32 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m), where n is the number of rows and m is the number of columns in the matrix
+# Space complexity: O(n), where n is the number of elements in the matrix.
```

Comment:
Actually I'd say this is O(n) where n is the number of rows or columns in the matrix.  You're not building an additional 2D array.